cicecore/cicedyn/general/ice_step_mod.F90

```diff
@@ -676,7 +676,7 @@ subroutine step_therm2 (dt, iblk)
          if (tmask(i,j,iblk)) then
 
          ! significant wave height for FSD
-         if (tr_fsd) &
+         if (tr_fsd .and. wave_spectrum(i,j,1,iblk) < 1.0+10_dbl_kind) &
```

Comment:
Is wave_spectrum always nonnegative?

Reply:
@eclare108213  - yes, it should be always non-negative.

Reply:
I'm pretty sure. @lettie-roach can confirm or deny this. The main thing is that in CESM at least, it assigns a missing value of 1.0e30 which is different than the typical coupler missing value of 1.0e36. I think we need to fix this eventually either in the WAVEWATCH III cap.

Reply:
Shouldn't we define a `ww3_missing_value` parameter in the mean time, and use that, instead of hardcoding this number here ?

Reply:
Perhaps. I am mainly just trying to catch any missing value above 1.0e10.

Reply:
Looks like wave_spectrum is always allocated (alloc_arrays_column) and but NOT always set (depends on tr_fsd).  So the above logic could cause problems with a subset of compilers and debug checks.  It would be safer to implement this using two if statements

if (tr_fsd) then
   if (wave_spectrum(i,j,1,iblk) < 1.0+10_dbl_kind) &
      wave_sig_ht(i,j,iblk) = c4*SQRT(SUM(wave_spectrum(i,j,:,iblk)*dwavefreq(:)))
endif